### PR TITLE
Add initial support for new versions TLS protocol.

### PR DESCRIPTION
**Pull request by raniervf**
Gmail do not accept TLSv1 anymore.
Need add new versions of TLS protocol (TLSv1_1, TLSv1_2 and TLSv1_3)

### CODE REVIEW

File: src/data-types/mailstream_cfstream.c [hunk 1]
```diff
@@ -972,6 +972,15 @@ int mailstream_cfstream_set_ssl_enabled(mailstream * s, int ssl_enabled)
       case MAILSTREAM_CFSTREAM_SSL_LEVEL_TLSv1:
         CFDictionarySetValue(settings, kCFStreamSSLLevel, kCFStreamSocketSecurityLevelTLSv1);
         break;
+      case MAILSTREAM_CFSTREAM_SSL_LEVEL_TLSv1_1:
```
Comment (dinhviethoa) [inline src/data-types/mailstream_cfstream.c:975]: Those constants are not defined anywhere.
You need to add them.

Comment (dinhviethoa) [reply src/data-types/mailstream_cfstream.c:975]: Ah, I see that's https://github.com/dinhviethoa/libetpan/pull/374

Comment (raniervf) [reply src/data-types/mailstream_cfstream.c:978]: Fix order to reflect enum type declaration.

**Comment by dinhviethoa** [comment]
Reverted: kCFStreamSocketSecurityLevelTLSv1_* is not defined in the system headers.

**Comment by jpstotz** [comment]
@dinhvh The official statement from Apple is just to use the strings:
> Important: The socket security level constants shown in Listing 1 are not yet defined in <CFNetwork/CFSocketStream.h> (r. 10229865) . For the moment you should use a literal string, as illustrated by the listing.
https://developer.apple.com/library/archive/technotes/tn2287/_index.html

**Pull request merged**